### PR TITLE
add user_agent_extra to 'ecs deploy' clients

### PR DESCRIPTION
**Pull request by allisaurus**
To identify calls made by `ecs deploy`

--
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/unit/customizations/ecs/test_ecsclient.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        # set global args
+        self.global_args = Namespace()
+        self.global_args.region = 'us-east-1'
+        self.global_args.endpoint_url = None
+        self.global_args.verify_ssl = None
+
+    def test_client_config(self):
+        self.test_client = ECSClient(
+            self.session, None, self.global_args, ECSDeploy.USER_DATA_EXTRA)
```
Comment (stealthycoin) [inline tests/unit/customizations/ecs/test_ecsclient.py:37]: I would change the `ECSDeploy.USER_DATA_EXTRA` constant here to be hardcoded in the test. The way it is now if the value got changed accidentally in the variable definition then the test would pass without complaint and the value would be wrong.

File: awscli/customizations/ecs/deploy.py [hunk 2]
```diff
@@ -399,13 +402,17 @@ def validate_deployment_group(self):
 
 
 class ECSClient():
-    def __init__(self, session, parsed_args, parsed_globals):
+
+    def __init__(self, session, parsed_args, parsed_globals, user_data_extra):
         self._args = parsed_args
+        self._custom_config = config.Config(
+            **{'user_agent_extra': user_data_extra})
```
Comment (stealthycoin) [inline awscli/customizations/ecs/deploy.py:409]: Same comment as above.

File: awscli/customizations/ecs/deploy.py [hunk 3]
```diff
@@ -109,22 +109,25 @@ class ECSDeploy(BasicCommand):
     MSG_SUCCESS = ("Successfully deployed {task_def} to "
                    "service '{service}'\n")
 
+    USER_DATA_EXTRA = 'customization/ecs-deploy'
```
Comment (stealthycoin) [inline awscli/customizations/ecs/deploy.py:112]: `USER_AGENT_EXTRA` here instead?

**Comment by allisaurus** [comment]
thanks for taking a look @stealthycoin ! this has been updated, per feedback. Let me know if we need anything else to push this over the finish line.

**Pull request merged**